extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobNameTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Marco Primo
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DestinationBlobNameTest {
+    DestinationBlobName destinationBlobName = new DestinationBlobName();
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName,  null, 1, blobName",
+            "partName, blobName,  null, 2, partName",
+            "partName, blobName,  '', 1, blobName",
+            "partName, blobName,  '', 2, partName",
+    }, nullValues = { "null" })
+    void shouldReturnBlobName_whenFolderNameIsNullOrEmpty(String partName, String blobName, String folderName, int partSize, String expected) {
```

Comment:
the name does not reflect the inputs (for case number 2 and 4 it is returning "partName", not blob name, I would split this test in two, one that verifies what happens when partSize is 1 and one with partSize greater than 1